webapp/channels/src/components/channel_layout/channel_identifier_router/channel_identifier_router.tsx

```diff
@@ -20,13 +20,13 @@ export interface Match {
 
 export type MatchAndHistory = Pick<Props, 'match' | 'history'>
 
-interface Props {
+type Props = {
```

Comment:
whats the reason for this changes?

Reply:
The new version of TS complains whenever you export a type that uses an unexported interface. In most of these cases, it's because the component is either exported as a `React.Component<UnexportedProps>` or `function (props: UnexportedProps): JSX.Element`. My VS Code has actually been complaining about that for months, so I've fixed it on files I've touched in other PRs, but I needed to do it on the rest of the files now.

The options for fixing it are to either export the interface or change it into a type, although I'm still not entirely sure why changing it into a type works. I chose to change them to types because most libraries I've looked at also used types for `Props`